release.sh

```diff
@@ -0,0 +1,464 @@
+
+  ${blue}${me} -c release 3.3.0${reset}
+    Creates a new Release on GitHub based on the tag 3.3.0 and uploads
+    the binary package based on master.
+    ${red}IMPORTANT: this will create a new tag if tag 3.3.0 doesn't exist,
+    so make sure to bump_version beforehand.${reset}
+
+  ${blue}${me} -v 3.3.0${reset}
+    Releases 3.3.0 in verbose mode.
+
+  ${blue}${me} -s 3.3.0${reset}
+    Simulates a 3.3.0 release: prints the commands instead of running them.
+
+EOF
+}
+
+function invalid_usage {
+  printf $red
+  echo $@
+  echo "Aborted"
+  printf $reset
+  show_help
+  exit -1;
+}
+function error_message {
+  printf $red
+  echo $@
+  echo "Aborted"
+  printf $reset
+  exit -1
+}
+function message {
+  if [[ $verbose == 1 ]]; then
+    printf $green
+    echo $@
+    printf $reset
+  fi
+}
+function run_message {
+  if [[ $simulate == 1 ]]; then
+    printf $yellow
+    echo $@
+    printf $reset
+  fi
+}
+
+#----------------
+# Read parameters
+#----------------
+
+# A POSIX variable
+# Reset in case getopts has been used previously in the shell.
+OPTIND=1
+
+# Read options:
+verbose=0
+simulate=0
+selected_cmd='all'
+
+while getopts "hvc:s" opt; do
+    case "$opt" in
+    h|\?)
+        show_help
+        exit 0
+        ;;
+    v)  verbose=1
+        ;;
+    s)  simulate=1
+        ;;
+    c)  selected_cmd="${OPTARG}"
+        ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+# Read argument
+version=$1
+
+# Validates arguments
+if [[ $2 ]]; then
+  invalid_usage "Invalid parameters"
+fi
+
+if [[ ! $( echo $version | grep -Ee '^[0-9]+\.[0-9]+(\.[0-9]+)?$' ) ]]; then
+  invalid_usage "Invalid version provided"
+fi
+
+message "Releasing version: $version"
+message "Running in verbose mode"
+if [[ $simulate == 1 ]]; then
+  message "Running in simulation mode"
+fi
+
+#---------------------------------
+# Check if we have the right tools
+#---------------------------------
+
+if ! type "git" > /dev/null; then
+  error_message "git not found, please install git before continuing: http://git.org"
+else
+  message "Found git: $(git --version)"
+fi
+
+if ! type "js" > /dev/null; then
+  error_message "SpiderMonkey interpreter not found, please install SpiderMonkey before continuing"
+else
+  message "Found SpiderMonkey"
+fi
+
+#------------------------------------
+# Check if we are in the right folder
+#------------------------------------
+if [[ ! -e '.git/config' ]]; then
+  error_message "You should run this command from the root directory of your repository."
+fi
+if [[ ! $( cat .git/config | grep -i 'automattic/facebook-instant-articles-wp') ]]; then
+  error_message "You should run this command from the root directory of the facebook-instant-articles-wp repository."
+fi
+
+repo_dir=$(pwd)
+
+#-------------------
+# Manages simulation
+#-------------------
+function run {
+  if [[ $simulate == 1 ]]; then
+    run_message $@
+  else
+    "$@"
+  fi
+}
+
+function revert_repo {
+  run cd $repo_dir
```

Comment:
I am not sure but I think this might halt if there are uncommitted files in the working directory when this method is called